### PR TITLE
[#IOPID-1910, #IOPID-1911] pdv integration for profiles and service preferences

### PR DESCRIPTION
**Pull request by arcogabbo**
#### List of Changes

* [[#IOPID-1910] PDV integration](https://github.com/pagopa/io-functions-elt/commit/ef6d37f0bd780c473597ad68af8be3fe118432e2)
* [added event tracking on errors](https://github.com/pagopa/io-functions-elt/commit/71d922b8a3323f5035abef9e01bdb2eb3d54d5e2)
* [added PDV BASE_URL env variable](https://github.com/pagopa/io-functions-elt/commit/985aba49491e29d846fcb9691b104b6a5b3381d4)
* [added unit tests for the enricher](https://github.com/pagopa/io-functions-elt/commit/0fe682349de37c1a5df98a3eddf1c610adf5222a)

#### Motivation and Context

add pdv integration

#### How Has This Been Tested?

unit tests, integration test with an event-hub and cosmosdb development resources

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: outbound/adapter/__tests__/pdv-id-outbound-enricher.test.ts [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+  const call = isList ? enricher.enrichs(value) : enricher.enrich(value);
+
+  const expectSuccess = (
+    result:
+      | readonly Result<MaybePdvDocumentsTypes>[]
+      | E.Either<Error, MaybePdvDocumentsTypes>
+  ) => {
+    if (isList) {
+      expect(result).toMatchObject([{ success: true }, { success: true }]);
```
Comment (gquadrati) [inline outbound/adapter/__tests__/pdv-id-outbound-enricher.test.ts:51]: We can check element in list, too

```suggestion
      expect(result).toMatchObject([
        { success: true, document: { ...value[0], userPDVId: aMockPdvId } },
        { success: true, document: { ...value[1], userPDVId: aMockPdvId } }
      ]);
```

**Review by gquadrati — APPROVED** [review]
good work! 🚀
I just left two small comments

File: utils/pdv.ts [hunk 2]
```diff
@@ -1,17 +1,67 @@
...
+              errors
+            )}`
+          )
+        )
+      )
+    ),
+    TE.mapLeft(error => {
+      // unexpected response that needs tracking
+      appInsightsTelemetryClient.trackEvent({
+        name: "fn-elt.error.unexpected",
```
[the remark this answers is not in the document]
Comment (arcogabbo) [reply utils/pdv.ts]: renamed in https://github.com/pagopa/io-functions-elt/pull/87/commits/0c4d2e0d2acf397d7ec1b02f07837ab6777dbfc1

File: outbound/adapter/__tests__/pdv-id-outbound-enricher.test.ts (hunk 1 above)
Comment (arcogabbo) [reply outbound/adapter/__tests__/pdv-id-outbound-enricher.test.ts]: check added in https://github.com/pagopa/io-functions-elt/pull/87/commits/0c4d2e0d2acf397d7ec1b02f07837ab6777dbfc1, thank you

**Pull request merged**